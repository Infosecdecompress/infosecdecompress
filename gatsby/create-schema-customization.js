const createSchemaCustomization = ({ actions }) => {
  const { createTypes } = actions;
  const typeDefs = `
    type MarkdownRemark implements Node {
      frontmatter: Frontmatter
    }
    type SocialImage {
      publicURL: String
    }
    type Frontmatter {
      socialImage: SocialImage
    }
  `;

  createTypes(typeDefs);
};

module.exports = createSchemaCustomization;