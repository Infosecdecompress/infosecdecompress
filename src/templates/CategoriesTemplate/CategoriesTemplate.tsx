import React from "react";

import { Link } from "gatsby";

import { Layout } from "@/components/Layout";
import { Page } from "@/components/Page";
import { Sidebar } from "@/components/Sidebar";
import { useCategoriesList, useSiteMetadata } from "@/hooks";
import { toKebabCase } from "@/utils";

const CategoriesTemplate: React.FC = () => {
  const { title, subtitle } = useSiteMetadata();
  const categories = useCategoriesList();

  return (
    <Layout title={`Categories - ${title}`} description={subtitle}>
      <Sidebar />
      <Page title="Categories">
        <ul>
          {categories.map(category => (
            <li key={category.fieldValue}>
              <Link to={`/category/${toKebabCase(category.fieldValue)}/`}>
                {category.fieldValue} ({category.totalCount})
              </Link>
            </li>
          ))}
        </ul>
      </Page>
    </Layout>
  );
};

export default CategoriesTemplate;
