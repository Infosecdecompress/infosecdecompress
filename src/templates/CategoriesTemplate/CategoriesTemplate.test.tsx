import React from "react";
import renderer from "react-test-renderer";

import { StaticQuery, useStaticQuery } from "gatsby";

import * as mocks from "@/mocks";

import CategoriesTemplate from "./CategoriesTemplate";

const mockedStaticQuery = StaticQuery as jest.Mock;
const mockedUseStaticQuery = useStaticQuery as jest.Mock;

describe("CategoriesTemplate", () => {
  beforeEach(() => {
    const props = {
      ...mocks.siteMetadata,
      allMarkdownRemark: mocks.allMarkdownRemark,
    };

    mockedStaticQuery.mockImplementationOnce(({ render }) => render(props));
    mockedUseStaticQuery.mockReturnValue(props);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<CategoriesTemplate />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
