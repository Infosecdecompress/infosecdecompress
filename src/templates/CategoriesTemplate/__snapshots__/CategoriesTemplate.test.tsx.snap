// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesTemplate renders correctly 1`] = `
<div>
  <div>
    <div>
      <div>
        <a
          href="/"
        />
        <div>
          <h2>
            <a
              href="/"
            >
              John Doe
            </a>
          </h2>
        </div>
        <p>
          InfosecDecompress.
        </p>
      </div>
      <nav>
        <ul>
          <li>
            <a
              href="/"
            >
              Articles
            </a>
          </li>
          <li>
            <a
              href="/pages/about"
            >
              About Me
            </a>
          </li>
          <li>
            <a
              href="/pages/contacts"
            >
              Contact Me
            </a>
          </li>
        </ul>
      </nav>
      <div
        style={
          {
            "margin": "5px auto",
          }
        }
      >
        <div
          style={
            {
              "margin": "0 auto",
            }
          }
        >
          <form
            onSubmit={[Function]}
          >
            <div
              style={
                {
                  "margin": "0 auto",
                }
              }
            >
              <input
                id="Search"
                onChange={[Function]}
                placeholder="搜尋文章"
                style={
                  {
                    "border": "1px solid #ccc",
                    "borderRadius": "4px",
                    "boxSizing": "border-box",
                    "display": "inline-block",
                    "margin": "0 auto",
                    "padding": "5px 5px",
                    "width": "250px",
                  }
                }
                value=""
              />
            </div>
          </form>
          <div>
            <table
              id="result"
              style={
                {
                  "border": "1px solid #ccc",
                  "borderCollapse": "collapse",
                  "borderRadius": "4px",
                  "display": "none",
                  "width": "250px",
                }
              }
            >
              <tbody />
            </table>
          </div>
        </div>
      </div>
      <div>
        Content is licensed under CC BY-SA 4.0
      </div>
    </div>
  </div>
  <div>
    <div>
      <h1>
        分類 標籤
      </h1>
      <div>
        <h2>
          分類
        </h2>
        <ul>
          <li>
            <a
              href="/category/typography/"
            >
              typography
               (
              1
              )
            </a>
          </li>
          <li>
            <a
              href="/category/design-inspiration/"
            >
              design inspiration
               (
              1
              )
            </a>
          </li>
        </ul>
        <h2>
          標籤
        </h2>
        <ul>
          <li>
            <a
              href="/tag/typography/"
            >
              typography
               (
              1
              )
            </a>
          </li>
          <li>
            <a
              href="/tag/design-inspiration/"
            >
              design inspiration
               (
              1
              )
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
